pinot-core/src/main/java/org/apache/pinot/core/operator/filter/FilterOperatorUtils.java

```diff
@@ -165,6 +179,9 @@ int getPriority(BaseFilterOperator filterOperator) {
         if (filterOperator instanceof OrFilterOperator) {
           return 4;
         }
+        if (filterOperator instanceof NotFilterOperator) {
+          return getPriority((BaseFilterOperator) filterOperator.getChildOperators().get(0));
```

Comment:
(minor) We can add a `getChildFilterOperator()` in `NotFilterOperator` which can save an unnecessary array allocation